Add unique_id attribute detail (#563)
home-assistant/developers.home-assistant: docs/entity_registry_index.md

sidebar_label: Introduction
---

> The entity registry is a registry where Home Assistant keeps track of entities. Any entity that is added to Home Assistant and has a unique ID (the `unique_id` attribute) will be registered in the registry.

Reviewer: The unique ID attribute being called `unique_id` is uh, obvious? I don't know if that adds anything. You can reword it maybe 
```suggestion
The entity registry is a registry where Home Assistant keeps track of entities. Any entity that is added to Home Assistant which specifies the [`unique_id` attribute](https://developers.home-assistant.io/docs/core/entity#generic-properties) will be registered in the registry.
```

Author: @balloob - No, that's the whole point to documentation - explaining what's not obvious to new developers.
I thought I already had a unique ID because my entity_ids were unique in Home Assistant. This page doesn't currently say how to assign a unique ID. What if the property required were `uniqueid` or `entity_id` or... ? I don't know until someone tells me.
The link to the generic properties is great, because there was no connection I could see in the docs between these pages. Nice.

> The entity registry is a registry where Home Assistant keeps track of entities. Any entity that is added to Home Assistant which specifies the [`unique_id` attribute](https://developers.home-assistant.io/docs/core/entity#generic-properties) will be registered in the registry.

Reviewer 2: We should probably use an internal link rather than external to be consistent with other pages
```suggestion
The entity registry is a registry where Home Assistant keeps track of entities. Any entity that is added to Home Assistant which specifies the [`unique_id` attribute](/core/entity.md#generic-properties) will be registered in the registry.
```

- latitude/longitude

If a device has a single serial but provides multiple entities, combine the serial with unique identifiers for the entities. For example, if a device measures both temperature and humidity, you can uniquely identify the entities using `{serial}-{sensor_type}`.

```python

> # Example, inside a component

Reviewer: This is not necessary if we link to the docs for entity properties above

Author: Sure. I see this section is more of an overview, but when I found and read it, I could not tell what to do. Will you add the link to entity properties, @balloob or how do we proceed?

Reviewer: I put that in my suggestion in the other comment?

    @property
    def unique_id(self):
        """Return the ID of this entity."""
        return self._unique_idty

> ```

Reviewer 2: This seems a little disconnected from the example outlined in the previous line. Perhaps either move up to line 7 (adding an explanation in line 6) or even better expand this example to show prefixing an entity with the serial

Author: @TomBrien - Now that Paulus has added a link to the other docs that show `entity_id` explicitly, we can remove this example and reduce the maintenance burden. Can you propose a change that takes this away so I can simply commit/accept it please? Then we're done, I believe. Thanks.

Reviewer 2: Agreed :)

Author: @TomBrien - I might not have been clear... I would like you to make the edit like you did for another one so I can click "Commit suggestion" to get rid of this. Is that OK?

Reviewer 2: Sorry my bad here you go
```suggestion
```
